velox/common/base/BitUtil.cpp

```diff
@@ -99,6 +99,10 @@ std::string toString(const uint64_t* bits, int offset, int size) {
   return ans;
 }
 
+void toString(const uint64_t* bits, int offset, int size, char* out) {
+  std::strcpy(out, (bits::toString(bits, offset, size)).c_str());
```

Comment:
I meant the reverse, you create this new `toString` without allocation, then reuse it in the existing one, something like this:

```c++
void toString(const uint64_t* bits, int offset, int size, char* out) {
  for (int i = 0; i < size; ++i) {
    out[i] = '0' + isBitSet(bits, offset + i);
  }
}

std::string toString(const uint64_t* bits, int offset, int size) {
  std::string ans(size);
  toString(bits, offset, size, ans.data());
  return ans;
}
```

Reply:
changes done